crates/nu-parser/src/parser.rs

```diff
@@ -1074,26 +1074,40 @@ pub fn parse_binary(
     working_set: &mut StateWorkingSet,
     span: Span,
 ) -> (Expression, Option<ParseError>) {
-    pub fn decode_hex(s: &str) -> Result<Vec<u8>, ParseIntError> {
-        (0..s.len())
-            .step_by(2)
-            .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
-            .collect()
+    let (hex_value, err) = parse_binary_with_base(working_set, span, 16, 2, b"0x[", b"]");
+    if err.is_some() {
+        return parse_binary_with_base(working_set, span, 2, 8, b"0b[", b"]");
     }
+    (hex_value, err)
+}
 
+fn parse_binary_with_base(
+    working_set: &mut StateWorkingSet,
+    span: Span,
+    base: u32,
+    min_digits_per_byte: usize,
```

Comment:
Let's hope they get optimized out in release mode. The `u8::from_str_radix` signature is kind of weird with a base exceeding the valid result type and a smaller range of safe and valid characters.